packages/shared/src/mock/values.ts

```diff
@@ -1,189 +1,241 @@
+    [9, 20, 41],
+  );
+};
+
+const threeD_rgb = () =>
+  ndarray(
     [
       [
-        [2, 0],
-        [1.41, 1.41],
-        [1, 1.73],
-        [0, 2],
+        [255, 0, 0],
+        [0, 255, 0],
+        [0, 0, 255],
       ],
       [
-        [4, 0],
-        [2.82, 2.82],
-        [2, 3.46],
-        [0, 4],
+        [0, 255, 255],
+        [255, 0, 255],
+        [255, 255, 0],
       ],
       [
-        [1, 0],
-        [0.71, 0.71],
-        [0.5, 0.87],
-        [0, 1],
+        [0, 0, 0],
+        [128, 128, 128],
+        [255, 255, 255],
       ],
-    ],
-    [
+    ].flat(2),
+    [3, 3, 3],
+  );
+
+export const mockValues = {
+  oneD,
+  oneD_linear: () => ndarray(range1()),
+  oneD_cplx,
+  oneD_compound,
+  oneD_bool,
+  oneD_errors: () => ndarray(oneD().data.map((val) => Math.abs(val) / 10)),
+  oneD_str: () => ndarray(['foo', 'bar']),
+  twoD,
+  twoD_cplx: () =>
+    ndarray(
       [
-        [-1, 0],
-        [0.87, -0.5],
-        [0.92, -0.38],
-        [1, 0],
-      ],
+        [cplx(0, -5), cplx(-2.1, -2)],
+        [cplx(5, 0), cplx(-3, 0.1)],
+      ].flat(1),
+      [2, 2],
+    ),
+  twoD_compound: () => {
+    const arrOneD = oneD();
+    const arrOneDComplex = oneD_cplx();
+    return ndarray(
       [
-        [-4, 0],
-        [3.46, -2],
-        [3.7, -1.53],
-        [4, 0],
+        ...oneD_compound().data,
+        ...['Vanadium', 'Niobium', 'Tantalum', 'Silicon', 'Germanium'].map<
+          [string, number, number, boolean, H5WebComplex]
+        >((element, index) => [
+          element,
+          index * 10,
+          arrOneD.get(index),
+          index % 2 === 1,
+          arrOneDComplex.get(index),
+        ]),
       ],
+      [2, 5],
+    );
+  },
+  twoD_bool: () => {
+    const { data: dataOneDBool } = oneD_bool();
+    return ndarray(
+      dataOneDBool.flatMap((rowBool) =>
+        dataOneDBool.map((colBool) => (rowBool ? colBool : !colBool)),
+      ),
+      [10, 10],
+    );
+  },
+  twoD_errors: () => {
+    const arr = range1();
+    return ndarray(
+      range2().flatMap((offset) => arr.map((val) => Math.abs(val - offset))),
+      [20, 41],
+    );
+  },
+  threeD,
+  threeD_cplx: () =>
+    ndarray(
       [
-        [-8, 0],
-        [6.93, -4],
-        [7.39, -3.06],
-        [8, 0],
-      ],
-    ],
-  ],
-  threeD_rgb: rgb,
-  fourD_rgb: range(0, 10).map((i) =>
-    rgb.map((rgb1) =>
-      rgb1.map((rgb2) => rgb2.map((val) => Math.min(val + i * 20, 255))),
+        [
+          [cplx(2, 0), cplx(1.41, 1.41), cplx(1, 1.73), cplx(0, 2)],
+          [cplx(4, 0), cplx(2.82, 2.82), cplx(2, 3.46), cplx(0, 4)],
+          [cplx(1, 0), cplx(0.71, 0.71), cplx(0.5, 0.87), cplx(0, 1)],
+        ],
+        [
+          [cplx(-1, 0), cplx(0.87, -0.5), cplx(0.92, -0.38), cplx(1, 0)],
+          [cplx(-4, 0), cplx(3.46, -2), cplx(3.7, -1.53), cplx(4, 0)],
+          [cplx(-8, 0), cplx(6.93, -4), cplx(7.39, -3.06), cplx(8, 0)],
+        ],
+      ].flat(2),
```

Comment:
Note that I'm always specifying the number of levels to flatten to ensure we get the correct `NdArray<...[]>` return type. It doesn't work as well with `Infinity`.